tool/tsh/db.go

```diff
@@ -869,6 +881,71 @@ func isMFADatabaseAccessRequired(cf *CLIConf, tc *client.TeleportClient, databas
 	return mfaResp.GetRequired(), nil
 }
 
+// checkRoute checks if a database route will be denied access.
+// Standard RBAC checks are still enforced
+// at connection time - this check just attempts to "fail fast" with
+// a user-friendly error.
+func checkRoute(tc *client.TeleportClient, cf *CLIConf, profile *client.ProfileStatus, dbRoute *tlsca.RouteToDatabase, db types.Database) (err error) {
+	roleSet, err := fetchCurrentClusterRoleSet(tc, cf, profile)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	dbUsers := roleSet.EnumerateDatabaseUsers(db)
+	// catch the cases where a user will be denied no matter what.
+	if dbUsers.WildcardDenied() {
+		return trace.AccessDenied("Your Teleport role(s) deny login as any db user (db_users wildcard %q is denied). Ask your Teleport administrator to grant you access. See: https://goteleport.com/docs/database-access/rbac/", types.Wildcard)
```

Comment:
> (db_users wildcard %q is denied)

A user can be unaware about teleport roles setup/definition so returning teleport specific information like `db_users wildcard` `db_names RBAC is used for %q protocol` makes the error message more confused. 

What would you say about alway returning descriptive/generic message like in https://github.com/gravitational/teleport/pull/13824/files#diff-dae87d4d3799335d85a5fa6ff17cfcbfb30705ae145c9e67ef70286423676a48R905 ? 

We can hide detailed teleport messages behind `-d debug flag.

Reply:
I pushed a change before I saw this comment, what was your suggested generic message?

Reply:
I was thinking about removing returning diffrent messages per  `WildcardDenied` `IsAnyAllowed` `!dbUsers.IsAllowed` and have one generic message like: 
`Your Teleport role(s) do not allow login to database %q with database user %q ...`

Reply:
I like the specificity of the messages though. I updated them to remove the RBAC details at least, but I think it's more helpful to know specifically why they aren't allowed. If we wanted to be generic, we could just remove all this client side stuff and only tell them to ask their Teleport admin to check the audit log when they get denied by the remote service RBAC check.